### ISSUE TITLE
Why is this not reported ?

### ISSUE THREAD

**Issue by eddycharly** [gordonklaus/ineffassign#70]
Let's consider the code below:
```go
package main

import "fmt"

type inner struct {
    s string
}

type outer struct {
    inner []inner
}

func main() {
    o := outer{
        inner: []inner{{
            s: "a",
        }},
    }
    for _, i := range o.inner {
        i.s = "b"
    }
    fmt.Println(o)
}
```
Shouldn't this code be reported ?

**Comment by eddycharly** [comment]
> For example, assignments to struct fields are never marked as ineffectual.

Because of this maybe ?

**Comment by gordonklaus** [comment]
Yes, basically.  Assignments to struct fields and array elements are not checked.

I suppose it might not be too much work to check whether a struct/array is ever used (or escapes) after having a field/element assigned to.  But I don't plan to do it :)

**Comment by eddycharly** [comment]
Thanks for clearing it up !
Do you know a tool that could report the code above as potentially wrong ? (useless assignment, useless code, or kind of warning that could help drawing a developer attention ?)

**Comment by gordonklaus** [comment]
Maybe [staticcheck](https://staticcheck.io/)?

**Comment by eddycharly** [comment]
`staticcheck` didn't report anything :(
Feel free to close the issue.
